osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/firmware/UpdateFirmwareCommandExecutor.java

```diff
@@ -134,12 +121,9 @@ private void verify(final ImageTransfer transfer) throws OsgpException {
     }
   }
 
-  private List<FirmwareVersionDto> activate(
-      final DlmsConnectionManager conn, final DlmsDevice device, final ImageTransfer transfer)
-      throws OsgpException {
+  private void activate(final ImageTransfer transfer) throws OsgpException {
     if (transfer.imageIsVerified() && transfer.imageToActivateOk()) {
       transfer.activateImage();
-      return this.getFirmwareVersionsCommandExecutor.execute(conn, device, null);
```

Comment:
Now the `GetFirmwareVersionsCommandExecutor` is not called anymore with `null` as 3rd parameter. The execute method of this executor checks on `!= null`. Could this be removed?

Reply:
There seem to be some other places where null is provided to the method, so i think it is saver to leave check in getFirmwareVersionsCommandExecutor